crates/sui-rosetta/src/operations.rs

```diff
@@ -168,85 +177,185 @@ impl Operations {
         status: Option<OperationStatus>,
     ) -> Result<Vec<Operation>, Error> {
         Ok(match tx {
-            SuiTransactionKind::PaySui(tx) => Self::parse_pay_sui_operations(sender, tx, status),
-            SuiTransactionKind::Call(tx) => Self::parse_call_operations(sender, status, tx)?,
+            SuiTransactionKind::ProgrammableTransaction(pt) => {
+                Self::parse_programmable_transaction(sender, status, pt)?
+            }
             _ => vec![Operation::generic_op(status, sender, tx)],
         })
     }
 
-    fn parse_call_operations(
+    fn parse_programmable_transaction(
```

Comment:
@patrickkuo this is a rather extensive rewrite, but should hold up for now for transactions that look _exactly_ like the ones generated by the ProgrammableTransactionBuilder. 
If folks start mixing/matching commands in unique ways, I don't think it will hold up very well.